core/dbt/deprecations.py

```diff
@@ -108,6 +108,21 @@ class ModelsKeyNonModelDeprecation(DBTDeprecation):
     '''
 
 
+class BigQueryPartitionByStringDeprecation(DBTDeprecation):
+    _name = 'bq-partition-by-string'
+
+    _description = '''
+    As of dbt v0.16.0, the `partition_by` config in BigQuery accepts a
+    dictionary containing `field` and `data_type`.
+
+    You provided: {raw_partition_by}
+    dbt inferred: {inferred_partition_by}
+
+    For more information, see:
+        [dbt docs link]
```

Comment:
I made a placeholder guide here, to be completed before 0.16.0 is released. Can you add this link accordingly? https://docs.getdbt.com/docs/upgrading-to-0-16-0